### PR TITLE
Increase unit test coverage

### PR DESCRIPTION
**Pull request by alexashley**
- convert existing unit tests to Ginkgo
  - migrated everything but `pkg/collector/ecr_events_test.go`, which was started in #37 
- backfill tests to increase coverage in the `attester` packge
- add [`mock`](https://github.com/golang/mock) package and `go generate` directives for scaffolding mocks

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,6 +43,12 @@ fmt:
 vet:
 	go vet ./...
 
+unit:
+	go test -cover -tags unit ./...
+
+mockgen:
+	go generate ./...
```
Comment (mrparkers) [inline Makefile:50]: should we add some code here that checks for `mockgen` and installs it if it isn't found?  similar to what the `generate` make target does

File: Makefile [hunk 2]
```diff
@@ -41,6 +43,12 @@ fmt:
 vet:
 	go vet ./...
 
+unit:
```
Comment (mrparkers) [inline Makefile:46]: should we add the `mockgen` target here as a dependency?

**Comment by mrparkers** [comment]
also, I'm getting some errors locally when trying to run `mockgen`.  looking into it now

**Comment by mrparkers** [comment]
pushed some changes to fix the issues I was running into locally

File: Makefile (hunk 2 above)
Comment (alexashley) [reply Makefile:46]: I might have it misconfigured, but AFAICT with `go generate` the mocks are recreated every time which adds about 20 seconds to running tests. 

I imagine that time will increase, so I would lean towards it being a one-off as necessary

File: Makefile (hunk 1 above)
Comment (alexashley) [reply Makefile:60]: Yeah that would be useful. I added a check and `go get` right before the generate call, although I didn't make it a separate target like the `controller-gen` is.

**Pull request closed without merging**